tool/src/main.rs

```diff
@@ -54,12 +57,39 @@ fn main() {
         tas = Tas::new().unwrap();
         println!("TAS created successfully.");
     }
-
-    let script_file = env::args().skip(1).next().unwrap_or("tas.lua".to_string());
+    let mut script_file = String::new();
+    if env::args().collect::<Vec<String>>().len() == 1 {
+        let cur_dir = std::env::current_dir().unwrap();
+        let cur_dir = cur_dir.to_str().unwrap();
+        let result = nfd::open_file_dialog(Some("lua"), Some(cur_dir)).unwrap_or_else(|e| { panic!(e) });
+        match result {
+            Response::Okay(file_path) => script_file = Path::new(&file_path).file_name().unwrap().to_str().unwrap().to_string(),
+            Response::OkayMultiple(_) => panic!("Multiple files selected."),
+            Response::Cancel => println!("Cancelled file selection.")
+        };
+    } else {
+        script_file = env::args().skip(1).next().unwrap();
+    }
```

Comment:
Let me introduce you to the amazingness of rust using expressions everywhere ;) `if` and `match` are just expressions, which return their last expression as return value. This means that you can write stuff like this:
```rust
let foo = if bar {
    baz
} else {
    qux
};
```
You can combine that with the inner match to produce something like
```rust
let scrip_file = if ...
```